### PR TITLE
Executor: Remove `LegacyInstanceReuse` strategy

### PR DESCRIPTION
**Pull request by yjhmelody**
It seems the old strategy have been depracted more than one year. 
So maybe it's time to clean up old strategy for wasm executor.

---
polkadot address: 15ouFh2SHpGbHtDPsJ6cXQfes9Cx1gEFnJJsJVqPGzBSTudr

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3637575

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3637981

File: substrate/client/executor/src/integration_tests/linux.rs [hunk 1]
```diff
@@ -33,8 +33,7 @@ fn memory_consumption_compiled() {
 
 	if std::env::var("RUN_TEST").is_ok() {
 		memory_consumption(WasmExecutionMethod::Compiled {
-			instantiation_strategy:
-				sc_executor_wasmtime::InstantiationStrategy::LegacyInstanceReuse,
+			instantiation_strategy: sc_executor_wasmtime::InstantiationStrategy::PoolingCopyOnWrite,
```
Comment (bkchr) [inline substrate/client/executor/src/integration_tests/linux.rs:36]:
```suggestion
			instantiation_strategy: sc_executor_wasmtime::InstantiationStrategy::RecreateInstance,
```

**Review by bkchr — COMMENTED** [review]
Tests are failing, otherwise looks good.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3645152

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3647235

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3648317

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3647524

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3649386

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3653378

**Comment by yjhmelody** [comment]
Is test `memory_consumption_compiled` still useful. Seem could not work for other strategies.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3653521

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3654061

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3666253

**Comment by yjhmelody** [comment]
Hi @bkchr @koute  I have no idea to solve `memory_consumption_compiled` test. Seems only work fine for old strategy.
Should I just remove this test?

**Comment by koute** [comment]
> Hi @bkchr @koute I have no idea to solve `memory_consumption_compiled` test. Seems only work fine for old strategy. Should I just remove this test?

Yes. Nuke it. We don't have to test this as in non-legacy strategies we don't do our own memory clearing, and `wasmtime` itself guarantees this.

**Comment by yjhmelody** [comment]
@koute anything need to be done？

**Review by bkchr — APPROVED** [review]

**Review by koute — APPROVED** [review]
LGTM; thanks!

**Comment by yjhmelody** [comment]
@bkchr Can I get a tip by bot?  If not, I will issue a small tipper at openGov for myself.

**Pull request merged**